packages/x-date-pickers/src/internals/components/PickersModalDialog.tsx

```diff
@@ -65,9 +87,19 @@ export const PickersModalDialog = (props: React.PropsWithChildren<PickersModalDi
   } = props;
 
   const ActionBar = components?.ActionBar ?? PickersActionBar;
+  const Dialog = components?.Dialog ?? PickersModalDialogRoot;
+  const Transition = components?.MobileTransition ?? Fade;
 
   return (
-    <PickersModalDialogRoot open={open} onClose={onDismiss} {...DialogProps}>
+    <Dialog
+      open={open}
+      onClose={onDismiss}
+      {...componentsProps?.dialog}
+      TransitionComponent={Transition}
+      TransitionProps={componentsProps?.mobileTransition}
+      PaperComponent={components?.MobilePaper}
+      PaperProps={componentsProps?.mobilePaper}
```

Comment:
If we keep the `contentPaper` should it also apply on mobile?
```suggestion
      PaperComponent={components?.MobilePaper ?? components?.ContentPaper}
      PaperProps={componentsProps?.mobilePaper ?? componentsProps?.contentPaper}
```

Reply:
@LukasTy was in favor not to because the layout wanted are often different
I will explain better than me

Reply:
`PaperContent` could possibly be added to the mobile pickers as well.
It could be up to the user to decide when to display it or not using CSS if it is needed.